dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -807,25 +807,16 @@ public List<Workflow> allPublishedWorkflows(
         @ApiParam(value = "Should only be used by Dockstore versions < 1.14.0. Indicates whether to get a service or workflow") @DefaultValue("false") @QueryParam("services") boolean services,
         @ApiParam(value = "Which workflow subclass to retrieve. If present takes precedence over services parameter") @QueryParam("subclass") WorkflowSubClass subclass,
         @Context HttpServletResponse response) {
-        try {
-            // delete the next line if GUI pagination is not working by 1.5.0 release
-            int maxLimit = Math.min(Integer.parseInt(PAGINATION_LIMIT), limit);
-            final Class<Workflow> workflowClass = (Class<Workflow>) workflowSubClass(services, subclass);
-            List<Workflow> workflows = workflowDAO.findAllPublished(offset, maxLimit, filter, sortCol, sortOrder,
-                    workflowClass);
-            filterContainersForHiddenTags(workflows);
-            stripContent(workflows);
-            EntryDAO entryDAO = services ? serviceEntryDAO : bioWorkflowDAO;
-            response.addHeader("X-total-count", String.valueOf(entryDAO.countAllPublished(Optional.of(filter), workflowClass)));
-            response.addHeader("Access-Control-Expose-Headers", "X-total-count");
-            return workflows;
-        } catch (IllegalArgumentException e) {
-            LOG.error("Could not get published workflows due to invalid arguments. Error is " + e.getMessage(), e);
-            throw new CustomWebApplicationException("Could not get published workflows due to invalid arguments. "
-                    + "Error is " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
-        }
-
-
+        int maxLimit = Math.min(Integer.parseInt(PAGINATION_LIMIT), limit);
+        final Class<Workflow> workflowClass = (Class<Workflow>) workflowSubClass(services, subclass);
+        List<Workflow> workflows = workflowDAO.findAllPublished(offset, maxLimit, filter, sortCol, sortOrder,
+                workflowClass);
+        filterContainersForHiddenTags(workflows);
+        stripContent(workflows);
+        EntryDAO entryDAO = services ? serviceEntryDAO : bioWorkflowDAO;
```

Comment:
This was the way the code was before, and is an unfortunate vestige from before we added the `subclass` parameter to this handler.  Really, here, we should select the `entryDAO` based on the type of entry (`workflowClass`), which could be `Notebook`, amongst other things.  I think the code probably works as-is, though, and if you can confirm that, no need to change right now.

As for later, as I'm pondering for another ticket, in cases like these, we need to figure out whether it's really important to select the specific `EntryDAO` that corresponds to the actual identity of an `Entry`.  Alternately, we can dispense with that requirement, and define that certain specialized `EntryDAO` operations should always work on all entry types, so you can use any `EntryDAO` to perform them...